test/functional/p2p_invalid_tx.py

```diff
@@ -158,13 +157,71 @@ def run_test(self):
         with node.assert_debug_log(['mapOrphan overflow, removed 1 tx']):
             node.p2p.send_txs_and_test(orphan_tx_pool, node, success=False)
 
+        self.log.info('Test orphan with rejected parents')
         rejected_parent = CTransaction()
         rejected_parent.vin.append(CTxIn(outpoint=COutPoint(tx_orphan_2_invalid.sha256, 0)))
         rejected_parent.vout.append(CTxOut(nValue=11 * COIN, scriptPubKey=SCRIPT_PUB_KEY_OP_TRUE))
         rejected_parent.rehash()
         with node.assert_debug_log(['not keeping orphan with rejected parents {}'.format(rejected_parent.hash)]):
             node.p2p.send_txs_and_test([rejected_parent], node, success=False)
 
+        self.log.info('Test erase of orphan tx from peer')
+        with node.assert_debug_log(['Erased 100 orphan tx from peer=24']):
```

Comment:
This test seems dependent on the context/test order. Is it possible to make it order-independent and to test for the change itself?

Reply:
This variant, i.e., the previous test state reuse, just suggests the minimum diff.